llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -4370,6 +4375,81 @@ getFalkorUnrollingPreferences(Loop *L, ScalarEvolution &SE,
   }
 }
 
+static bool shouldUnrollLoopWithInstruction(Instruction &I,
+                                            AArch64TTIImpl &TTI) {
+  // Don't unroll vectorised loop.
+  if (I.getType()->isVectorTy())
+    return false;
+
+  if (isa<CallInst>(I) || isa<InvokeInst>(I)) {
+    if (const Function *F = cast<CallBase>(I).getCalledFunction())
+      if (!TTI.isLoweredToCall(F))
+        return true;
+    return false;
+  }
+
+  return true;
+}
+
+static InstructionCost getSizeOfLoop(Loop *L, AArch64TTIImpl &TTI) {
```

Comment:
Can this also be shared with `getAppleRuntimeUnrollPreferences`? Although there we explicitly don't bail out on vector instructions. We only bail out on loops LV already vectorized.

Reply:
Done!